pom.xml

```diff
@@ -728,8 +728,9 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.8.1</version>
         <configuration>
-          <source>${java.version}</source>
-          <target>${java.version}</target>
+          <source>15</source>
+          <target>15</target>
+	         <compilerArgs>--enable-preview</compilerArgs>
```

Comment:
Take a look at https://github.com/checkstyle/checkstyle/pull/8813 . Any input files that use Java syntax that is from a release that is greater than that of checkstyle must live in respective `non-compilable` directories. Take note that you must use the `getNonCompilablePath()` method to get the correct path in your unit tests. It would be good to add a completely new input file, with records declarations that include type parameters, varargs, etc. You can verify the compilation of the input file yourself using `javac`. Additionally, make sure that you include the header `//non-compiled with javac: Compilable with Java14 ` or whatever release you are targeting to your input file so that can be verified in CI.

Reply:
how do u guys know all these thing, i searched about it but couldnt find anything helpful
but thank you.

Reply:
send an update, ci will fail but i have send it to say one thing that i think this check right now only work if modifier is there.
so this is not acceptable?

Reply:
>  only work if modifier is there

Please share cli output with good and bad example.

Reply:
METHOD_DEF, RECORD_DEF and CTOR_DEF
for these 3?